src/transformers/generation/utils.py

```diff
@@ -1705,6 +1709,69 @@ def typeerror():
                 **model_kwargs,
             )
 
+    @torch.no_grad()
+    def heal_tokens(self, input_ids: torch.LongTensor) -> torch.LongTensor:
+        r"""
+
+        Generates sequences of token ids for models with a language modeling head.
+
+        Parameters:
+            input_ids (`torch.LongTensor`): The sequence used as a prompt for the generation.
+
+        Return:
+            `torch.LongTensor` where each sequence has its tail token replaced with its appropriate extension.
+        """
+
+        from pygtrie import CharTrie
+
+        from ..models.auto import AutoTokenizer
```

Comment:
These imports should go to the top. It also means `pygtrie` should be added to  [this list](https://github.com/huggingface/transformers/blob/de13a951b38b85195984164819f1ab05fe508677/setup.py#L97) ⚠️